storage/merge.go

```diff
@@ -462,11 +461,10 @@ func (c *chainSampleIterator) Seek(t int64) bool {
 }
 
 func (c *chainSampleIterator) At() (t int64, v float64) {
-	if len(c.h) == 0 {
-		panic("chainSampleIterator.At() called after .Next() returned false.")
+	if c.curr == nil {
```

Comment:
Why is this changing?

Reply:
True, not needed.

Reply:
Actually it's needed. It's because we now have current one out of heap.

Reply:
How does that help us? The top of the heap is always the one we want, so it seems like extra complexity.